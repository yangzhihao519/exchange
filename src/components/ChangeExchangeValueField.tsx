import React from 'react'
import Money from '../models/money'

interface ChangeExchangeValueFieldProps {
    exchange: Money,
    onInputChange: (value: number, isSellingOut: boolean) => void,
    isSellingOut: boolean
}

interface ChangeExchangeValueFieldState {}

class ChangeExchangeValueField extends React.Component<ChangeExchangeValueFieldProps, ChangeExchangeValueFieldState>{

    handleChange = (e: any) => {
        this.props.onInputChange(e.target.value, this.props.isSellingOut)
    }

    render() {
        let {isSellingOut, exchange} = this.props
        let inputValue = exchange.value

        return (<div> 
            {inputValue > 0 && <span>{isSellingOut ? "-" : "+"}</span>}
            <input type="number" value={inputValue} onChange={(e: any) => this.handleChange(e)} />
        </div>)
    }
    
}

export default ChangeExchangeValueField