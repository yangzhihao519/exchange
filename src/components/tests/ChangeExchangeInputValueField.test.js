import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ChangeExchangeValueField from '../ChangeExchangeValueField';

const mockFunction = jest.fn();

describe('ChangeExchangeValueField', () => {
  it('renders without crashing given the required props', () => {
    const props = {
        exchange: {},
        onInputChange: mockFunction,
        isSellingOut: false
    }
    const wrapper = shallow(<ChangeExchangeValueField {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})