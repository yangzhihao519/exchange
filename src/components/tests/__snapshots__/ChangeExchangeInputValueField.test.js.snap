// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeExchangeValueField renders without crashing given the required props 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
  />
</div>
`;
